paddle/phi/kernels/multi_tensor_adam_utility_kernel.cuh

```diff
@@ -0,0 +1,139 @@
+#include "paddle/phi/backends/gpu/gpu_context.h"
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/fluid/framework/tensor_util.h"
+#include "paddle/phi/core/tensor_utils.h"
+
+#include <assert.h>
+
+namespace phi {
+
+#define MAX_COMPUTE_GROUP_SIZE 65535
+
+constexpr int max_tensors_size[6] = {110, 64, 48, 36, 30, 24};
```

Comment:
I prefer not to fixing the sizes. The `max_tensors_size` and `max_blocks_size` would be set in the caller directly.